scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -895,6 +895,12 @@ def apply_filters(self, **kwargs):
                     if ts.arch_exclude and plat.arch in ts.arch_exclude:
                         instance.add_filter("In test case arch exclude", Filters.TESTSUITE)
 
+                    if ts.vendor_allow and plat.vendor not in ts.vendor_allow:
+                        instance.add_filter("Not in test suite vendor allow list", Filters.TESTSUITE)
+
+                    if ts.vendor_exclude and plat.vendor in ts.vendor_exclude:
```

Comment:
also has a trick with `plat.vendor`: when in testcase.yaml for some mistake/reason there is an empty `vendor_exclude` item, then all platforms without vendor specified will be excluded.

Reply:
if there is an empty `vendor_exclude`, we will not get into this conditional.

Reply:
hmm, I've tried it this way and it was filtered:
```
vendor_exclude:
 -
```

Reply:
well, you are adding an empty element, so there is a list with an empty element. The same "mistake" can happen anywhere, it is not something unique to vendor_exclude, try this with other keys and you will get different behaviours all over the place.